metadata-ingestion/src/datahub/ingestion/transformer/clear_dataset_ownership.py

```diff
@@ -0,0 +1,40 @@
+import datahub.emitter.mce_builder as builder
+from datahub.configuration.common import ConfigModel
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.transformer.dataset_transformer import DatasetTransformer
+from datahub.metadata.schema_classes import (
+    DatasetSnapshotClass,
+    MetadataChangeEventClass,
+    OwnershipClass,
+)
+
+
+class ClearDatasetOwnershipConfig(ConfigModel):
+    pass
+
+
+class SimpleClearDatasetOwnership(DatasetTransformer):
+    """Transformer that clears all owners on each dataset."""
+
+    def __init__(self, config: ClearDatasetOwnershipConfig, ctx: PipelineContext):
+        pass
+
+    @classmethod
+    def create(
+        cls, config_dict: dict, ctx: PipelineContext
+    ) -> "SimpleClearDatasetOwnership":
+        config = ClearDatasetOwnershipConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    def transform_one(self, mce: MetadataChangeEventClass) -> MetadataChangeEventClass:
+        if not isinstance(mce.proposedSnapshot, DatasetSnapshotClass):
+            return mce
+
+        ownership = builder.get_or_add_aspect(
```

Comment:
There could be cases where ingestion is not bringing in owners, but owners have also been added via UI. In that case, ownership metadata will be overwritten by the clearing transformer. It might be useful to set owners to [] only if ownership is already set to some value by ingestion.

Reply:
Correct. That is the only use case. This is similar to add dataset ownership transformer.

 e.g. in case of glue metadata I found some tables had ownership of `hadoop`. So I want to mark all of them as removed. I use them alongwith `add_dataset_ownership` to get the correct owners ingested. I can add a warning in the docs and the use case.

Reply:
Note that this is true for other transformers too. `add_dataset_tags` will actually set dataset tags in the UI. From a transformer only perspective the naming makes sense. As it is doing work on what is sent from the source. Added a warning for the same in the docs

Reply:
FYI: In the case of tags, when the UI attaches tags, those are added to EditableSchemaMetadata, so they are kept separate from the system ingested tags. 

I think it is okay to implement this as per the name (clearing ownership), but I'm not sure about whether this is practically what people would want to do during an ingestion. 

e.g. if you want to clear ownership for a bunch of datasets, it might be simpler to just issue REST calls to clear ownership for those dataset urns. 

Our story for sending "delta-s" for ownership is not great, so until we build that, we will encounter these issues. Hopefully we can fix up these transformers in the future to implement true delta capability. So, `add` and `remove` instead of `set` and `clear`.

Reply:
So if I ingest from this source periodically it will add the extra owner every time (because `hadoop` is set as owner by glue). And then run REST endpoint to remove owners every time. This will keep on increasing the owner aspect version every time for no reason for all the datasets.  

Clearing out owners during ingestion is a cleaner way instead of letting wrong data get in and cleaning it after ingestion.